File: Bozosort/Ruby/jcla1/bozosort.rb
# The bang (!) denotes that the function mutates the original array
# Works by, if not sorted yet, picking 2 random indicies
# and swapping the 2 elements
def bozosort!(arr)
    while not arr.sort == arr
        a, b = rand(arr.length), rand(arr.length)
        arr[a], arr[b] = arr[b], arr[a]
    end
    return arr
end

File: Bozosort/Go/jcla1/bozosort.go
package bozosort

import "math/rand"

func Bozosort(arr []int) []int {
    for !isSorted(arr) {
        a, b := rand.Intn(len(arr)), rand.Intn(len(arr))
        arr[a], arr[b] = arr[b], arr[a]
    }

    return arr
}

func isSorted(arr []int) bool {
    for i := 1; i < len(arr); i++ {
        if arr[i-1] > arr[i] {
            return false
        }
    }

    return true
}
